Describe using `AccessibilityContent` for elements and screens

karabatov opened the pull request:
As we can now use `AccessibilityContent` in automation, we should try and do that more, and rely less on copy, same for locating if a screen is displayed.

Also fixes some typos and header levels.

---

I'm interested in feedback re: `ScreenNaming`, as it's now used in home screen and notifications.

Commit:
Structure and typos

Commit:
Describe AccessibilityContent and ScreenNaming

ilyapuchka commented on Cookbook/Technical-Documents/AutomationIdentifiers.md:
````diff
@@ -67,23 +67,90 @@ fileprivate enum NavigationBar {
+    .firstMatch
+```
+
+Note that the identifier is usually set on the cell itself rather than its internal components, as automation has problems finding elements otherwise.
````
I don't think that happens always. It also probably makes sense to mention how the identifier then should be set on the cell in code.

```suggestion
Note that you may need to set the identifier on the cell itself rather than its internal components, as automation may have problems finding elements otherwise.
```

ilyapuchka commented on Cookbook/Technical-Documents/AutomationIdentifiers.md:
```diff
@@ -67,23 +67,90 @@ fileprivate enum NavigationBar {
+
+Note that the identifier is usually set on the cell itself rather than its internal components, as automation has problems finding elements otherwise.
+
+### Taking advantage of ScreenNaming for screen identifiers
```
I find this section very confusing and I'd prefer if it would only mention that Bento is doing it this way and how then it should be reflected in the automation. And a note that ScreenNaming should not be used for setting accessibility identifiers manually, instead it should use AccessibilityContent.

P.S. IMO Bento should not assign accessibility identifiers automatically especially using something from completely different domain (analytics) and instead offer a way to set accessibility identifier for the screen manually. But that's not a discussion about Bento.

karabatov replied:
Yes, probably makes sense to downplay it a little bit.

karabatov replied:
Following discussion on Slack, mention avoiding using `ScreenNaming` raw identifiers without setting them manually as accessibility identifiers, as automatic setting of them will likely be removed.

JasonDobo commented on Cookbook/Technical-Documents/AutomationIdentifiers.md:
````diff
@@ -67,23 +67,90 @@ fileprivate enum NavigationBar {
+
+``` swift
+let promofield = app.tables
+    .cells[AccessibilityContent.PromoCode.codeCell]
````
```suggestion
    .matching(.cell, identifier: AccessibilityContent.PromoCode.codeCell)
```

JasonDobo replied:
I generally don't like using ["string"] in a query as it will throw an acceptation if multiple. matches are found. The code I suggested will work regards of a single or multiple matches for a Cell with the identifier

karabatov commented:
I'm going to address the comments soon 🙏

karabatov commented:
Following discussion on Slack, add a note about preferring `matching(identifier: …).firstMatch` to using subscripts for identifiers.

Commit:
Code review comments

Commit:
Replace the section about ScreenNaming

Commit:
Revert accidental change

karabatov commented:
@ilyapuchka @JasonDobo @inamiy I have made the requested changes:
- [x] Changed `[…]` to `.matching(…)` for elements (cells).
- [x] Made a note about using `matching(…).firstMatch` rather than `[…]`.
- [x] Replaced the section on `ScreenNaming` with using `Screen` accessibility identifiers in automation instead (after a Slack debate).
